### ISSUE TITLE
DirectX 8 games no longer working?

### ISSUE THREAD

**Issue by yawnbunny** [IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT#528]
### Checklist [README]

- [X] Device is not undervolted nor overclocked
- [X] Device is using the [latest drivers](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/wiki/Intel-driver-updates-and-changelogs)
- [X] Game is not cracked, modded and use the latest patch

### Game / Application [Required]

Any D3D8 game (Postal 2, Arma: Cold War Assault, Max Payne)

### Game Platform [Required]

- [X] Steam
- [ ] Microsoft Store
- [ ] GOG Galaxy
- [ ] Ubisoft Connect
- [ ] Battle.net
- [ ] Epic Games
- [ ] Rockstar Games Launcher
- [ ] EA app
- [ ] Other (please specify)

### Processor / Processor Number [Required]

Ryzen 5600

### Graphic Card [Required]

Arc A750

### GPU Driver Version [Required]

- [X] 31.0.101.4826
- [X] 31.0.101.4824
- [ ] 31.0.101.4676
- [ ] 31.0.101.4672
- [ ] 31.0.101.4669
- [ ] 31.0.101.4644
- [ ] 31.0.101.4578
- [ ] 31.0.101.4577
- [ ] 31.0.101.4576
- [ ] 31.0.101.4575
- [ ] 31.0.101.4514
- [ ] 31.0.101.4502
- [ ] 31.0.101.4499
- [ ] 31.0.101.4382
- [ ] 31.0.101.4369
- [ ] 31.0.101.4335
- [ ] 31.0.101.4314
- [ ] 31.0.101.4311
- [ ] 31.0.101.4257
- [ ] 31.0.101.4255
- [ ] 31.0.101.4148
- [ ] 31.0.101.4146
- [ ] 31.0.101.4125
- [ ] 31.0.101.4123
- [ ] 31.0.101.4091
- [ ] 31.0.101.4090
- [ ] 31.0.101.4034

### Rendering API [Required]

- [ ] Vulkan
- [ ] OpenGL
- [ ] DirectX12
- [ ] DirectX11
- [ ] DirectX10
- [X] DirectX9

### Windows Build Number [Required]

- [X] Windows 11 22H2
- [ ] Windows 11 21H2
- [ ] Windows 10 22H2
- [ ] Windows 10 21H2
- [ ] Other (Please specify)

### Intel System Support Utility report

If you can't replicate by just booting up any of the games I mentioned on 4824 on an A750/770 and there's a chance it's setup specific I'll add it.

### Description and steps to reproduce [Required]

With 4824 and 4826 DirectX 8 games don't work. Nothing more to say. I go back to 4672 and it works fine. I haven't tested 4676. 
I've tested Postal 2, Arma: Cold War Assault and Max Payne. Max Payne plays the intro FMV but crashes on the menu. The other two don't boot. 
Even with a dirty driver install without reboots or DDU it's the same, 4672 works and 4824 doesn't. I can (and do) use various wrappers like DGVoodoo, D3D8to9, DX8VK, but it'd be awfully nice if I didn't have to use them for games that worked fine as is.
![MXoa5u3cQG](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/assets/134241612/f4ee4540-96b1-41c0-b749-9b181c6b5ec5)
![k5oHl89rvL](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/assets/134241612/7064369f-4716-4d51-9727-d66d0d4c6327)

### Game graphic quality [Required]

- [X] Low
- [X] Medium
- [X] High
- [X] Ultra

### Game resolution [Required]

2560x1440

### Game VSync [Required]

On

### Game display mode [Required]

- [X] Fullscreen
- [X] Windowed
- [X] Borderless Windowed

### Detailed game settings [Required]

Doesn't matter.

**Comment by Karen-Intel** [comment]
Hello @yawnbunny help needed!

Can you please change the title of this game and select one of the 3 you have reported + I need you to generate another 2 threads with the other 2 games.

Thank you, I am reporting this asap but I need your help with it to create a case per game.

Karen

**Comment by yawnbunny** [comment]
Done. I'll make another for Postal 2 since these two are the most relevant and run the most issue free.

**Comment by BiZkViT96** [comment]
Can you check, if you seeing the same behavior in [3DMark2001 SE](https://benchmarks.ul.com/downloads/3dmark2001.exe) (KEY: 7QMF2-TSLJR-ESRDD-L3FJF) with 4672 vs 4826 driver. Is a DirectX 8 benchmark and it uses Remedy MAX-FX engine.

Ignore MMX warning, main window should open anyway.

**Comment by yawnbunny** [comment]
@BiZkViT96 
Can't even finish the setup on 4826, fails at texture conversion. It installed and did a lap in the game just fine on 4672. 
On 4826, gives me the same "setFullscreen - creating the device failed : D3DERR_INVALIDCALL" seen in the Max Payne error box screenshot in the original post.
![huipM91JCq](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/assets/134241612/076c0b49-85a0-4ae7-bff0-be5d37cf0611)
![C7knmSqwbz](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/assets/134241612/13169240-5218-4dbc-af20-0d932f48df95)

**Comment by BiZkViT96** [comment]
Try patches from UCyborg. 

For 3DMark 
[3DMarkStartupHangPatchv1.01.zip](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/files/12679889/3DMarkStartupHangPatchv1.01.zip)

For Max Payne 
[MaxPayneStartupHangPatchv1.01.zip](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/files/12679890/MaxPayneStartupHangPatchv1.01.zip)

Obviously if all DX8 games is affected then this is not a solution to the problem and it should be solved on driver side.

**Comment by yawnbunny** [comment]
Yeah, that's my point. All D3D8 games are affected, not a single one boots on 4824/4826, when they all boot on 4672.

**Comment by Eschryn** [comment]
Gothic 2 is also affected. Game apparently also runs using DX8. So the regression is seemingly part of this issue.

**Comment by Karen-Intel** [comment]
Hey @yawnbunny just FYI leaving here internal tracking number: 22018875778
Our driver cadence for fixes takes ~ 3 to 6 months but I have confirmed the regression to driver v.4676 so it is an option to play this game and others related to this behavior. Stay tuned for updates on this thread, I will post any update I get so please keep it open.

Karen

**Comment by Eschryn** [comment]
Another alternative may be trying to run it on Linux - Gothic runs really well on Linux under Proton

**Comment by yawnbunny** [comment]
I forgot to update but the issue remains on 4885.

**Comment by paulo-jgf** [comment]
Issue remains on 4887. RivaTuner Statistics Server throws a D3D8 error on startup. I am positive everything DX8 is not working right:

![image](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/assets/40891723/1537c7ec-bd9d-46ec-9a46-7454aea01ed0)

Ryzen 5600 Asrock B450 (bios AGESA Combo V2 PI 1.2.0.7) Asrock A770 8GB W10-22H2

**Comment by yawnbunny** [comment]
4952 is the newest, but yes it still remains and nothing D3D8 runs unless you use a wrapper like DGVoodoo, DX8VK or D3D8to9.
